soc/riscv/riscv-privileged/common/soc_common_irq.c

```diff
@@ -16,29 +16,54 @@
 
 #if defined(CONFIG_RISCV_HAS_CLIC)
 
+void arch_irq_intc_enable(const struct device *intc, unsigned int irq)
+{
+	ARG_UNUSED(intc);
+	iscv_clic_irq_enable(irq);
+}
+
 void arch_irq_enable(unsigned int irq)
 {
-	riscv_clic_irq_enable(irq);
+	arch_irq_intc_enable(NULL, irq);
 }
 
-void arch_irq_disable(unsigned int irq)
+void arch_irq_intc_disable(const struct device *intc, unsigned int irq)
 {
+	ARG_UNUSED(intc);
 	riscv_clic_irq_disable(irq);
 }
 
-int arch_irq_is_enabled(unsigned int irq)
+void arch_irq_disable(unsigned int irq)
 {
+	arch_irq_intc_disable(NULL, irq);
+}
+
+int arch_irq_intc_is_enabled(const struct device *intc, unsigned int irq)
```

Comment:
the order of parameters should be switched. That way, the contents of registers do not need to be swapped.

Reply:
This has never been in my consideration before 🤯